CONTRIBUTING.md

```diff
@@ -78,6 +79,24 @@ Signed-off-by: Jane Smith <jane.smith@email.com>
 ```
 You may type this line on your own when writing your commit messages. However, if your user.name and user.email are set in your git configs, you can use `-s` or `--signoff` to add the `Signed-off-by` line to the end of the commit message.
 
+## Changelog
+
+Flow framework maintains version specific changelog by enforcing a change to the ongoing [CHANGELOG](CHANGELOG.md) file adhering to the [Keep A Changelog](https://keepachangelog.com/en/1.0.0/) format.
+
+Briefly, the changes are curated by version, with the changes to the main branch added chronologically to `Unreleased` version. Further, each version has corresponding sections which list out the category of the change - `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `Security`.
+
+
+### How to add my changes to [CHANGELOG](CHANGELOG.md)?
+
+As a contributor, you must ensure that every pull request has the changes listed out within the corresponding version and appropriate section of [CHANGELOG](CHANGELOG.md) file.
```

Comment:
I think "must" is a bit too strong.  I'd really like a bit longer paragraph here covering:
 - substantive changes _should_ be listed
 - deprecations _must_ be listed
 - trivial / typo / doc fixes do not need to be listed.   In these cases add the skip-changelog tag.

No need to invent something, I'm content with whatever is in widespread use in the org.